src/views/connectionUI.ts

```diff
@@ -347,7 +347,7 @@ export class ConnectionUI {
 					connectFunc = this.fillOrPromptForMissingInfo(selection);
 				}
 
-				connectFunc.then((resolvedConnectionCreds) => {
+				await connectFunc.then((resolvedConnectionCreds) => {
```

Comment:
FYI - the await here doesn't really do anything since this whole thing is dependent on the resolve being called (it doesn't really matter when the inner function itself completes). 

It'll still technically work correctly but what was there before was just fine.